From 7408f046057f91daf5c78262d5ea9069bc843213 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 1 Sep 2018 11:11:17 +0200
Subject: [PATCH] gluon-mesh-batman-adv: respondd: remove incorrect field width
 limits

At least the ifindex and the flags fields can be larger than 0xff.

Fixes #1523
---
 package/gluon-mesh-batman-adv/src/respondd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/gluon-mesh-batman-adv/src/respondd.c b/package/gluon-mesh-batman-adv/src/respondd.c
index ca7ae96..ff3972d 100644
--- a/package/gluon-mesh-batman-adv/src/respondd.c
+++ b/package/gluon-mesh-batman-adv/src/respondd.c
@@ -94,7 +94,7 @@ static struct json_object * get_addresses(void) {
 		if (sscanf(line,
 			   "%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
 			   "%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
-			   "  %*2x %*2x %*2x %2x %"STRINGIFY(IF_NAMESIZE)"s",
+			   "  %*x %*x %*x %x %"STRINGIFY(IF_NAMESIZE)"s",
 			   &addr.s6_addr[0], &addr.s6_addr[1], &addr.s6_addr[2], &addr.s6_addr[3],
 			   &addr.s6_addr[4], &addr.s6_addr[5], &addr.s6_addr[6], &addr.s6_addr[7],
 			   &addr.s6_addr[8], &addr.s6_addr[9], &addr.s6_addr[10], &addr.s6_addr[11],
-- 
2.5.0

